.github/workflows/main.yml

```diff
@@ -0,0 +1,13 @@
+name: CI
+
+on: [pull_request, workflow_dispatch]
```

Comment:
I would also suggest to add push on main.
so maybe add:

`push:
         branches: [main]`

Reply:
Due to our settings the main can not be modified without a pull request.

Reply:
How does pull against master behave?
Is it seen like a push to push when merged?

Reply:
If I understand you correctly, then yes.